pkg/build/api/v1/conversion.go

```diff
@@ -212,9 +212,17 @@ func convert_v1_BuildStrategy_To_api_BuildStrategy(in *BuildStrategy, out *newer
 
 func addConversionFuncs(scheme *runtime.Scheme) {
 	err := scheme.AddDefaultingFuncs(
+		func(source *BuildSource) {
+			if (source != nil) && (source.Type == BuildSourceBinary) {
+				// initialize the Binary source to an empty default state if type but no value set
+				if source.Binary == nil {
```

Comment:
you are right. 

should I fix that ? (and in the other defaulting functions below, too) ?